addon/adapters/pouch.js

```diff
@@ -131,7 +133,8 @@ export default DS.RESTAdapter.extend({
 
   _recordToData: function (store, type, record) {
     var data = {};
-    var serializer = store.serializerFor(type.typeKey);
+    var recordTypeName = this.getRecordTypeName(type);
+    var serializer = store.serializerFor(recordTypeName);
```

Comment:
That's a good point. I initially used `type.modelName || type.typeKey` instead of `getRecordTypeName` in this method for that reason. But the serialized data comes back with a camelized key (see line 155 below), and I figured it wouldn't hurt to use the camelized version in both places.

ED has supported either dasherized or camelized names for lookups (`modelFor`, `serializerFor`, etc) for a while — have you seen something that indicates that it is going to stop supporting camelized ones at some point?